Tools/Scripts/build-and-collect-pgo-profiles

```diff
@@ -28,18 +48,61 @@ mkdir -p "$BASE/motionmark"
 mkdir -p "$BASE/output"
 mkdir -p "$BASE/Internal/WebKit/WebKitAdditions/Profiling/"
 
-cd ../Internal
+cd Internal
+
+if [[ -z $APP ]] ; then
+    echo "Building WebKit..."
+    rm -rf ../OpenSource/WebKitBuild
+    make release WK_LTO_MODE=thin ENABLE_LLVM_PROFILE_GENERATION=ON
+else
+    if [[ ! -d "$APP" ]] ; then
+        echo "$APP is missing, aborting."
+        DisplayHelp
+        exit
+    fi
+    echo "Using .app: $APP"
+fi
+
+jsargs=(
+    --plan jetstream2
+    --diagnose-directory="$BASE/jetstream"
+    --generate-profiles
+    --count 1
+)
 
-rm -rf ../OpenSource/WebKitBuild
-make release WK_LTO_MODE=thin ENABLE_LLVM_PROFILE_GENERATION=ON
+spargs=(
+    --plan speedometer2
+    --diagnose-directory="$BASE/speedometer"
+    --generate-profiles
+    --count 1
+)
+
+mmargs=(
+    --plan motionmark1.1
+    --diagnose-directory="$BASE/motionmark"
+    --generate-profiles
+    --count 1
+)
+
+if [[ -n $APP ]] ; then
+   # add --browser-path
+   jsargs+=(--browser-path "$APP")
+   spargs+=(--browser-path "$APP")
+   mmargs+=(--browser-path "$APP")
+else
+   # add --build-directory
+   jsargs+=(--build-directory ../OpenSource/WebKitBuild/Release)
+   spargs+=(--build-directory ../OpenSource/WebKitBuild/Release)
+   mmargs+=(--build-directory ../OpenSource/WebKitBuild/Release)
```

Comment:
This is basically hard-coding a Mac release build. Is that deliberate? Maybe we should make this a variable since the string is somewhat long.

Reply:
This was again from the existing code, but I think a variable would be a good idea in case there are other use cases in the future. I've added a new flag to specify a build directory.